### PR TITLE
syncer(dm): handle `;` in SQL comment when load table structure

### LINKED ISSUE #6205: DM should handle escape db&table name from dump files
### What did you do?

create table with escape name. see https://github.com/pingcap/dumpling/issues/131

### What did you expect to see?

DM parse db&table as normal

### What did you see instead?

DM parse failed.
https://github.com/pingcap/tiflow/blob/06039a56f58400be5657138159396ee7421f6aa7/dm/pkg/utils/file.go#L141-L163

### Versions of the cluster

DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):
```console
(paste DM version here, and you must ensure versions of dmctl, DM-worker and DM-master are same)
```
Upstream MySQL/MariaDB server version:
```console
(paste upstream MySQL/MariaDB server version here)
```
Downstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):
```console
(paste TiDB cluster version here)
```
How did you deploy DM: tiup or manually?
```console
(leave TiUP or manually here)
```
Other interesting information (system version, hardware config, etc):
```console
>
>
```

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: close #6205

### What is changed and how it works?

except for `\n` in identifier, we have confidence to treat `;\n` as the end of a statement. We hope user will not use `\n` in identifier.

#### Tests 

 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by amyangfei — COMMENTED** [review]
> We hope user will not use \n in identifier.

Does it mean DM still doesn't support uesr to create a table with name like `t;\n2`. 😂

**Comment by lance6716** [comment]
> > We hope user will not use \n in identifier.
> 
> Does it mean DM still doesn't support uesr to create a table with name like `t;\n2`. 🤔 @lance6716

yes. i guess loader of DM will also not work for this case. not sure about lightning.

File: dm/tests/downstream_more_column/data/db1.prepare.sql [hunk 1]
```diff
@@ -2,6 +2,10 @@ drop database if exists `downstream_more_column`;
 drop database if exists `downstream_more_column1`;
 create database `downstream_more_column1`;
 use `downstream_more_column1`;
-create table t1 (c1 int, c2 int, c3 int, primary key(c1));
+create table t1 (c1 int, c2 int, c3 int comment '1;
```
Comment (GMHDBJD) [inline dm/tests/downstream_more_column/data/db1.prepare.sql:5]: Can we test with dumpling format, this stmt will be format by ShowCreateTable as
```
CREATE TABLE `t1` (
  `c1` int(11) NOT NULL,
  `c2` int(11) DEFAULT NULL,
  `c3` int(11) DEFAULT NULL COMMENT '1;\n2',
  PRIMARY KEY (`c1`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1
```

**Pull request merged**